File: features/person/src/main/java/com/nivasafinance/features/person/exception/PersonOperationException.java
package com.nivasafinance.features.person.exception;

import com.nivasafinance.common.exception.BadRequestException;
import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

import java.io.Serial;

public class PersonOperationException extends BadRequestException {
    @Serial
    private static final long serialVersionUID = 882423L;
    public PersonOperationException(String messageCode, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(messageCode, null, messageSource));
    }
}

File: features/person/src/main/java/com/nivasafinance/features/person/exception/PersonOperationException.kt
package com.nivasafinance.features.person.exception

import com.nivasafinance.common.exception.ExceptionUtils
import org.springframework.context.MessageSource

class PersonOperationException(
    messageKey: String,
    messageSource: MessageSource
) : RuntimeException(
    ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource)
)
